pkg/health/server/server.go

```diff
@@ -99,35 +105,72 @@ func (s *Server) DumpUptime() string {
 	return time.Since(s.startTime).String()
 }
 
-// getNodes fetches the latest set of nodes in the cluster from the Cilium
-// daemon, and updates the Server's 'nodes' map.
-func (s *Server) getNodes() (nodeMap, error) {
+// getNodes fetches the nodes added and removed from the last time the server
+// made a request to the daemon.
+func (s *Server) getNodes() (nodeMap, nodeMap, error) {
 	scopedLog := log
 	if s.CiliumURI != "" {
 		scopedLog = log.WithField("URI", s.CiliumURI)
 	}
-	scopedLog.Debug("Sending request for /healthz ...")
+	scopedLog.Debug("Sending request for /cluster/nodes ...")
 
-	resp, err := s.Daemon.GetHealthz(nil)
+	clusterNodesParam := daemon.NewGetClusterNodesParams()
+	clusterNodesParam.SetClientID(&s.clientID)
+
+	resp, err := s.Daemon.GetClusterNodes(clusterNodesParam)
 	if err != nil {
-		return nil, fmt.Errorf("unable to get agent health: %s", err)
+		return nil, nil, fmt.Errorf("unable to get nodes' cluster: %s", err)
 	}
-	log.Debug("Got cilium /healthz")
+	log.Debug("Got cilium /cluster/nodes")
 
-	if resp == nil || resp.Payload == nil || resp.Payload.Cluster == nil {
-		return nil, fmt.Errorf("received nil health response")
+	if resp == nil || resp.Payload == nil {
+		return nil, nil, fmt.Errorf("received nil health response")
 	}
 
-	if resp.Payload.Cluster.Self != "" {
+	s.clientID = resp.Payload.ClientID
```

Comment:
This is still not protected by `s.RWMutex`, so `getNodes()` should not be called concurrently right?

Reply:
@joestringer Correct, is that the wrong assumption?

Reply:
I think it's strictly correct today, but it's a little weird to:
* Not tell in the function comment that it is only safe for single-threading, AND
* Use mutex for protecting some fields but not others (see a few lines down, the `self` handling)

I'd expect that if mutexes are used inside the function, then the writers have thought through all the multithreading concerns and I don't have to worry about it when using the function.

To alleviate this concern, we could just leave a comment above the function so someone reading later can atleast read that and ask the question, "wait, why does this have to be single-threaded?"

Alternatively, `resp.Payload.Self` will *always* be set, so we're already taking the RWMutex, so just extend the RWMutex around this call as well?